src/rust/src/backend/kdf.rs

```diff
@@ -1696,11 +1697,317 @@ impl ConcatKdfHmac {
     }
 }
 
+// NO-COVERAGE-START
+#[pyo3::pyclass(
+    module = "cryptography.hazmat.primitives.kdf.kbkdf",
+    name = "KBKDFHMAC"
+)]
+// NO-COVERAGE-END
+struct KbkdfHmac {
+    algorithm: pyo3::Py<pyo3::PyAny>,
+    length: usize,
+    params: KbkdfParams,
+    used: bool,
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(PartialEq)]
+enum CounterLocation {
+    BeforeFixed,
+    AfterFixed,
+    MiddleFixed,
+}
+
+struct KbkdfParams {
+    rlen: usize,
+    llen: Option<usize>,
+    location: CounterLocation,
+    label: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    context: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    fixed: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    break_location: Option<usize>,
+}
+
+#[allow(clippy::too_many_arguments)]
+fn validate_kbkdf_parameters(
+    py: pyo3::Python<'_>,
+    mode: pyo3::Py<pyo3::PyAny>,
+    rlen: usize,
+    llen: Option<usize>,
+    location: pyo3::Py<pyo3::PyAny>,
+    label: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    context: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    fixed: Option<pyo3::Py<pyo3::types::PyBytes>>,
+    break_location: Option<usize>,
+) -> CryptographyResult<KbkdfParams> {
+    let mode_bound = mode.bind(py);
+    let mode_type = crate::types::KBKDF_MODE.get(py)?;
+    if !mode_bound.is_instance(&mode_type)? {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyTypeError::new_err("mode must be of type Mode"),
+        ));
+    }
+
+    let location_bound = location.bind(py);
+    let counter_location = crate::types::KBKDF_COUNTER_LOCATION.get(py)?;
+    if !location_bound.is_instance(&counter_location)? {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyTypeError::new_err("location must be of type CounterLocation"),
+        ));
+    }
+
+    let counter_location_before_fixed =
+        counter_location.getattr(pyo3::intern!(py, "BeforeFixed"))?;
+    let counter_location_after_fixed = counter_location.getattr(pyo3::intern!(py, "AfterFixed"))?;
+    let rust_location = if location_bound.eq(&counter_location_before_fixed)? {
+        CounterLocation::BeforeFixed
+    } else if location_bound.eq(&counter_location_after_fixed)? {
+        CounterLocation::AfterFixed
+    } else {
+        // There are only 3 options so this is MiddleFixed
+        CounterLocation::MiddleFixed
+    };
+
+    if rust_location == CounterLocation::MiddleFixed && break_location.is_none() {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err("Please specify a break_location"),
+        ));
+    }
+
+    if break_location.is_some() && rust_location != CounterLocation::MiddleFixed {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err(
+                "break_location is ignored when location is not CounterLocation.MiddleFixed",
+            ),
+        ));
+    }
```

Comment:
`break_location` should probably be an element inn `MiddleFixed` rather than a separate Option